ftp-source/src/main/java/org/springframework/cloud/stream/module/ftp/FtpSource.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.ftp;
+
+import java.util.Collections;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableModule;
+import org.springframework.cloud.stream.annotation.ModuleChannels;
+import org.springframework.cloud.stream.annotation.Source;
+import org.springframework.cloud.stream.module.MaxMessagesConfigurationProperties;
+import org.springframework.cloud.stream.module.PeriodicTriggerConfiguration;
+import org.springframework.cloud.stream.module.file.FileConsumerConfigurationProperties;
+import org.springframework.cloud.stream.module.file.FileReadingMode;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.SourcePollingChannelAdapterSpec;
+import org.springframework.integration.dsl.core.Pollers;
+import org.springframework.integration.dsl.ftp.Ftp;
+import org.springframework.integration.dsl.support.Consumer;
+import org.springframework.integration.file.splitter.FileSplitter;
+import org.springframework.integration.file.transformer.FileToByteArrayTransformer;
+import org.springframework.integration.ftp.session.DefaultFtpSessionFactory;
+import org.springframework.integration.scheduling.PollerMetadata;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.scheduling.Trigger;
+
+/**
+ * @author David Turanski
+ */
+@EnableModule(Source.class)
+@EnableConfigurationProperties({FtpSourceConfigurationProperties.class,
+		FileConsumerConfigurationProperties.class, MaxMessagesConfigurationProperties.class})
+@Import({PeriodicTriggerConfiguration.class, FtpSessionFactoryConfiguration.class})
+public class FtpSource {
+
+	@Autowired
+	private FtpSourceConfigurationProperties config;
```

Comment:
This follows the style of similar examples.  Usually with one ConfigurationProperties set, it is called 'config'. Here we have more than one, so `config` denotes the primary one.